### PR TITLE
Open up develop branch for non-stable commits.

### PR DESCRIPTION
**Pull request by Dead2**
Start 2.1.0 develop cycle.
Replace ZLIBNG_VERNUM with a larger one with space for more accurate information.
Replace ZLIBNG_VER_SUBREVISION with ZLIBNG_VER_MODIFIED and ZLIBNG_VER_STATUS.

### CODE REVIEW

File: zlib.h
Comment (mtl1979) [inline zlib.h]: Should be "non-zero".

File: zlib-ng.h
Comment (mtl1979) [inline zlib-ng.h:52]: Might make more sense to have `ZLIBNG_VER_MODIFIED` as last element in version string.

Comment (Dead2) [reply zlib-ng.h:52]: Yes, perhaps so.
I considered this back and forth when making the change, and both ways makes sense for different cases.
It kind of depends on what kind of modification we are talking, a patched version made by a distro? Makes sense to put it last. A more serious fork of zlib-ng? Might make sense to put it where it is now or even before major.

There really is no good way of mixing modified (from a fork) releases with official releases, for that it'd need to be an epoch thing due to the potential for API/ABI differences. Should we support such a thing? Probably not.

Comment (mtl1979) [reply zlib-ng.h:52]: Personally I think we should not "support" more serious forks... Those should be handled the same way as difference between stock zlib and zlib-ng.

Comment (mtl1979) [reply zlib-ng.h:52]: API/ABI breakakes can happen for various reasons... What matters is the question if the change will get upstreamed to zlib-ng or not... if it gets, then the change modifies `ZLIBNG_VER_MINOR` and any change to `ZLIBNG_VER_MODIFIED` is irrelevant.

This is my opinion as former package maintainer for RedHat.

**Comment by Dead2** [comment]
Changed example.c to always print out the zlib-ng version, as it would likely not make sense to build it against stock zlib. Fixes the issue with the new ZLIBNG_VERNUM differing in length compared to the zlib one.

**Comment by mtl1979** [comment]
@Dead2 I was kinda waiting for you to notice that issue... I didn't have any rush as I wanted to wait for CI to finish to see if there is any new warnings...

**Comment by nmoinvaz** [comment]
LGTM. I like that you have kept the version number define and went with OpenSSL format. minizip-ng also uses openSSL format.

**Review by nmoinvaz — APPROVED** [review]

**Review by mtl1979 — APPROVED** [review]
LGTM.

**Pull request merged**
